bcycle-backend/Controllers/ShareController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Threading.Tasks;
+using bcycle_backend.Controllers.Utils;
+using bcycle_backend.Models;
+using bcycle_backend.Models.Responses.Views;
+using bcycle_backend.Security;
+using bcycle_backend.Services;
+using Microsoft.AspNetCore.Mvc;
+
+namespace bcycle_backend.Controllers
+{
+    [ApiController]
+    [Route("/api/share")]
+    public class ShareController : ControllerBase
+    {
+        private readonly ShareService _shareService;
+        private readonly TripService _tripService;
+
+        public ShareController(ShareService shareService, TripService tripService)
+        {
+            _shareService = shareService;
+            _tripService = tripService;
+        }
+
+        // GET /api/share/group-trip/:sharingId
+        [HttpGet("group-trip/{sharingId:guid}")]
+        public async Task<ActionResult<ResultContainer<GroupTripShareView>>> GetGroupTrip(Guid sharingId)
+        {
+            var groupTripInfo = await _shareService.GetGroupTripAsync(sharingId);
+            if (groupTripInfo == null) return NotFound();
+            return new ResultContainer<GroupTripShareView>(groupTripInfo);
+        }
+
+        // GET /api/share/trip/:tripId
+        [HttpGet("trip/{sharingId:guid}")]
+        public async Task<ActionResult<ResultContainer<PrivateTripShareView>>> GetPrivateTrip(Guid sharingId)
+        {
+            var privateTripInfo = await _shareService.GetPrivateTripAsync(sharingId);
+            if (privateTripInfo == null) return NotFound();
+            return new ResultContainer<PrivateTripShareView>(privateTripInfo);
+        }
+
+        // POST /api/share/:tripId
```

Comment:
Why not leave this in `TripsController`?

Reply:
The truth is, i wasn't completely sure about this change, like 55:45. It is sharing related as well as trip related, so responsibility wise it fits both. On the other hand trip controller had a lot more methods that share controller did at this point. Also putting that in share controller would make the url prettier. Taking all of it into account i decided to go for the `ShareController`